aptos-move/framework/aptos-framework/sources/reconfiguration.move

```diff
@@ -149,6 +153,8 @@ module aptos_framework::reconfiguration {
                 epoch: config_ref.epoch,
             },
         );
+
+        reconfiguration_state::on_reconfig_finish();
```

Comment:
this doesn't call all the on_new_epoch update functions? I thought we want to use the buffer for both cases and just make sure it's applied on_new_epoch

Reply:
I feel `reconfiguration_state` has to be at the very bottom of the dependency graph and can't call them (otherwise circular dependency).

Here's the full graph.

![image](https://github.com/aptos-labs/aptos-core/assets/4213569/9ac4bbea-0aba-4369-9fb7-718d64b86053)